go.mod

```diff
@@ -1,6 +1,6 @@
 module go.uber.org/goleak
 
-go 1.18
+go 1.20
```

Comment:
why? I thought this is now not recommended, as it forces people to upgrade to Go 1.20?

Reply:
This is ok. We only support the last 2 versions of Go releases.

Reply:
Yeah, that was the intent. Go 1.20 is the minimum supported version right now so that's what I moved the directive to. Allows use of 1.20 features and semantics if needed.

Reply:
Yeah, I understand.

But while pre 1.21 bumping this didn't really matter, post 1.21 it actively breaks people, right? So updating this*without using any new language semantics* seems rude.

Reply:
Yeah, that's a reasonable point. If it happens to work with an older version, this will force an upgrade there as well post 1.21. (This specific change is still pre-1.21 so it won't.) I've reverted this line nonetheless.

Reply:
Funny note/update: I'm going to have to bump it to 1.20 in another PR today because I want errors.Join.